ompi/mpi/fortran/mpif-h/request_get_status_some_f.c

```diff
@@ -0,0 +1,140 @@
+#pragma weak mpi_request_get_status_some__ = ompi_request_get_status_some_f
+
+#pragma weak MPI_Request_get_status_some_f = ompi_request_get_status_some_f
+#pragma weak MPI_Request_get_status_some_f08 = ompi_request_get_status_some_f
+#else
+#if ! OMPI_BUILD_MPI_PROFILING
+OMPI_GENERATE_F77_BINDINGS (MPI_REQUEST_GET_STATUS_SOME,
+                           mpi_request_get_status_some,
+                           mpi_request_get_status_some_,
+                           mpi_request_get_status_some__,
+                           ompi_request_get_status_some_f,
+                           (MPI_Fint *incount, MPI_Fint *array_of_requests, MPI_Fint *outcount, MPI_Fint *array_of_indices, MPI_Fint *array_of_statuses, MPI_Fint *ierr),
+                           (incount, array_of_requests, outcount, array_of_indices, array_of_statuses, ierr) )
+#else
+#define ompi_request_get_status_some_f pompi_request_get_status_some_f
+#endif
+#endif
+
+
+static const char FUNC_NAME[] = "MPI_REQUEST_GET_STATUS_SOME";
+
+
+void ompi_request_get_status_some_f(MPI_Fint *incount, MPI_Fint *array_of_requests,
+                    MPI_Fint *outcount, MPI_Fint *array_of_indices,
+                    MPI_Fint *array_of_statuses, MPI_Fint *ierr)
+{
+    int c_ierr;
+    MPI_Request *c_req;
+    MPI_Status  *c_status;
+    int i;
+    OMPI_SINGLE_NAME_DECL(outcount);
+    OMPI_ARRAY_NAME_DECL(array_of_indices);
+
+    /* Shortcut to avoid malloc(0) if *count==0.  We're intentionally
+       skipping other parameter error checks. */
+    if (OPAL_UNLIKELY(0 == OMPI_FINT_2_INT(*incount))) {
+        *outcount = OMPI_INT_2_FINT(MPI_UNDEFINED);
+        *ierr = OMPI_INT_2_FINT(MPI_SUCCESS);
+        return;
+    }
+
+    c_req = (MPI_Request *) malloc(OMPI_FINT_2_INT(*incount) *
+                   (sizeof(MPI_Request) + sizeof(MPI_Status)));
+    if (NULL == c_req) {
+        c_ierr = OMPI_ERRHANDLER_NOHANDLE_INVOKE(
+                                        MPI_ERR_NO_MEM,
+                                        FUNC_NAME);
+        if (NULL != ierr) *ierr = OMPI_INT_2_FINT(c_ierr);
+        return;
+    }
+    c_status = (MPI_Status*) (c_req + OMPI_FINT_2_INT(*incount));
+
+    for (i = 0; i < OMPI_FINT_2_INT(*incount); ++i) {
+        c_req[i] = PMPI_Request_f2c(array_of_requests[i]);
+    }
+
+    OMPI_ARRAY_FINT_2_INT_ALLOC(array_of_indices, *incount);
+    c_ierr = PMPI_Request_get_status_some(OMPI_FINT_2_INT(*incount), c_req,
+                                          OMPI_SINGLE_NAME_CONVERT(outcount),
+                                          OMPI_ARRAY_NAME_CONVERT(array_of_indices),
+                                          c_status);
+    if (NULL != ierr) *ierr = OMPI_INT_2_FINT(c_ierr);
+
+    if (MPI_SUCCESS == c_ierr) {
+        OMPI_SINGLE_INT_2_FINT(*outcount);
+        OMPI_ARRAY_INT_2_FINT(array_of_indices, *outcount);
+
+        /* Increment indexes by one for fortran conventions */
+
+        if (MPI_UNDEFINED != OMPI_FINT_2_INT(*outcount)) {
```

Comment:
Are you sure this is correct ? First you alter the `array_of_requests` which is technically an IN. Second, you populate the `array_of_statuses` according to `incount` and not to `outcount`. Here is what the MPI standard states about this:
> Returns in the first outcount locations of the array array_of_indices the indices of these operations within the array array_of_requests; the array is indexed from zero in C and from one in Fortran. Returns in the first outcount
locations of the array array_of_statuses the status for these completed operations.